framework/doc/content/source/problems/SolutionInvalidity.md

```diff
@@ -0,0 +1,47 @@
+
+To declare a solution as "invalid", use the macro in the following code excerpt with the message of your choice:
+
+!listing /test/src/materials/NonsafeMaterial.C  re=if \(_input_diffusivity.*\s.*flagInvalidSolution.*\;
+
+If any solution invalidity is detected during the solve, a summary table of solution invalid warnings will be generated and reported at the end of each time step as below:
+
+```
+Solution Invalid Warnings:
+-------------------------------------------------------------------------------------------------------
+|     Section     | Current | Timestep | Total |                        Message                       |
+-------------------------------------------------------------------------------------------------------
+| NonsafeMaterial |      16 |       64 |    64 | The diffusivity is greater than the threshold value! |
+-------------------------------------------------------------------------------------------------------
+
+```
+**section** shows the moose object name registered for the solution invalid detection.
+**Current** shows the number of solution invalid warnings for the latest iteration.
+**Timestep** shows the number of solution invalid warnings for one time step.
+**Total** shows the total number of soluton invalid warnings for the solve.
+**Message** shows the decription of the solution invalidity.
+
+This Solution Invalid Warning table can be silenced by setting [!param](/Problem/FEProblem/allow_invalid_solution) to 'true'. Then the converged solution will be allowed even there are still solution invalid warnings, but a message will be generated in the end of the calculation as a reminder:
+```
+*** Warning ***
+The Solution Invalidity warnings are detected but silenced! Use Problem/allow_invalid_solution=false to activate
+
+```
+
+The Solution Invalid Warning can also be printed out immediately after it is detected by setting [!param](/Problem/FEProblem/immediately_print_invalid_solution) to `true`.
+
+The two functions used internally in the 'SolutionInvalidInterface' for registeration are shown below:
+
+`flagInvalidSolutionInternal()` is an internal method used by the flagInvalidSolution macro that flags a solution as invalid with the given ID.
+
+`registerInvalidSolutionInternal()` is an internal method used by the flagInvalidSolution macro to statically register an invalid solution case.
+
+The code is showing below:
+
+!listing /framework/include/interfaces/SolutionInvalidInterface.h re=void flagInvalid.*\s*.*\;
```

Comment:
Let's remove the content about internal methods - we don't need users to know these.